src/main/java/org/hiero/sketch/table/LazyMembership.java

```diff
@@ -41,7 +40,7 @@ public LazyMembership(final IMembershipSet baseMap, final Predicate<Integer> fil
         if (filter == null) throw new NullArgumentException("PartialMembershipDense cannot be " +
                 "instantiated with a null filter");
         this.baseMap = baseMap;
-        this.rowCount = baseMap.getSize(false);
+        this.rowCount = 0;
```

Comment:
This is not about getSize, it's about the iterator itself.
There is the possibility, but the question is "who will do it".
The users of your classes only call these methods, all policies about data representation have to be part of your classes.